tests/tests.cpp

```diff
@@ -73,8 +73,10 @@ int main()
         try
         {
             LOG("Worker thread started, sleeping for a while...");
-            assert(a.mStr == "move test moved" && !strcmp(b, "test message")
-               && (c == -20));
+//  Thread might move the string more than once.

```

Comment:
I have been testing on both the win32 and posix versions of MinGW w64, as part of ensuring that the library doesn't cause conflicts. The multiple move happens when using winpthreads (due to the "use existing implementation" part of my contribution working as intended). I suspect it might also happen if the test is run on Visual Studio, though I have not tested that case.

Multiple moves are made apparent by your code with or without the assertion. Expect "move test moved moved moved" when using libwinpthreads.

Reply:
Okay, so this is related to the silent inclusion of <thread> if it is available. This assert was intended to test the efficiency of our implementation, not the standard lib. There was a PR from another user that raised the topic about the efficiency of passing the arguments, which was done via std::bind. With his help I managed to optimize that without using C++14 features. It seems that implementation is more efficient in that it does only a single move. Not sure if that would work on all compilers, so that strict assert is not a good idea
 But it raises another point - we have to be sure which implementation is actually being tested.